### PR TITLE
feat: state for webhook simulation

### PR DESCRIPTION
**Pull request by khaledmashaly**
## What does this PR do?

create webhook simulation state

### CODE REVIEW

File: packages/backend/src/app/database/database-connection.ts [hunk 1]
```diff
@@ -65,6 +67,7 @@ const getMigrations = () => {
         addEventRouting1678382946390,
         bumpFixPieceVersions1678928503715,
         migrateSchedule1679014156667,
+        CreateWebhookSimulationSchema1680698259291,
```
Comment (abuaboud) [inline packages/backend/src/app/database/database-connection.ts:70]: nit: can we make it small letter?

File: packages/backend/src/app/webhooks/webhook-simulation/webhook-simulation-service.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+type DeleteParams = BaseParams
+type GetParams = BaseParams
+type CreateParams = BaseParams
+
+type AcquireLockParams = {
+    flowId: FlowId
+    op: 'create' | 'delete'
+}
+
+const acquireLock = async ({ flowId, op }: AcquireLockParams): Promise<Lock> => {
```
Comment (abuaboud) [inline packages/backend/src/app/webhooks/webhook-simulation/webhook-simulation-service.ts]: Please update to the new lock.

**Review by abuaboud — DISMISSED** [review]

File: packages/backend/src/app/database/database-connection.ts (hunk 1 above)
Comment (khaledmashaly) [reply packages/backend/src/app/database/database-connection.ts:70]: This is a class name :sweat_smile:

File: packages/backend/src/app/webhooks/webhook-simulation/webhook-simulation-service.ts (hunk 2 above)
Comment (khaledmashaly) [reply packages/backend/src/app/webhooks/webhook-simulation/webhook-simulation-service.ts]: fixed in 8f8b286

**Pull request merged**